src/main/java/org/spongepowered/api/item/inventory/type/Inventory2D.java

```diff
@@ -72,23 +73,10 @@
         return getSlot(pos).map(Inventory::peek);
     }
 
-    /**
-     * Gets without removing the stack at the supplied position in this
-     * Inventory.
-     *
-     * @see Inventory#peek()
-     * @param pos Slot position to query
-     * @param limit item limit
-     * @return matching stacks, as per the semantics of {@link Inventory#peek()}
-     */
-    default Optional<ItemStack> peek(Vector2i pos, int limit) {
```

Comment:
Why remove this?

Reply:
see above: 
> removed peek with limit. (instead use query and totalQuantity)

Reply:
Then why keep `poll` with limit, if it can also be done with query and total quantity. How do you use "query and totalQuantity"?

Reply:
It works in all cases for "peek" because its readonly.
```
Inventory queryResult = inventory.query(ITEM_STACK_IGNORE_QUANTITY.of(stack));
int quantityOfStack = queryResult.totalQuantity();
```
But it won't work for polling a specific quantity as it also modifies the inventory.
Including only removing partial stacks.
```
Inventory queryResult = inventory.query(ITEM_STACK_IGNORE_QUANTITY.of(stack));
queryResult.poll(32);
```